components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -314,52 +314,73 @@ class Queue(BaseModel):
 class S3Credentials(BaseModel):
     access_key_id: str
     secret_access_key: str
+    session_token: Optional[str]
 
     @validator("access_key_id")
     def validate_access_key_id(cls, field):
-        if field == "":
+        if "" == field:
             raise ValueError("access_key_id cannot be empty")
         return field
 
     @validator("secret_access_key")
     def validate_secret_access_key(cls, field):
-        if field == "":
+        if "" == field:
             raise ValueError("secret_access_key cannot be empty")
         return field
 
 
+class AwsAuthentication(BaseModel):
+    type: Literal["ec2", "env_vars", "profile", "credentials"]
+    profile: Optional[str] = None
+    credentials: Optional[S3Credentials] = None
+
+    @root_validator(pre=True)
+    def validate_authentication(cls, values):
+        auth_type = values.get("type")
+        profile = values.get("profile")
+        credentials = values.get("credentials")
+
+        if "profile" == auth_type:
+            if not profile:
+                raise ValueError("profile must be set when type is 'profile'")
+            if credentials:
+                raise ValueError("credentials must not be set when type is 'profile'")
+        elif "credentials" == auth_type:
+            if not credentials:
+                raise ValueError("credentials must be set when type is 'credentials'")
+            if profile:
+                raise ValueError("profile must not be set when type is 'credentials'")
```

Comment:
nit: how about first check that credentials and profile are not both specfied outside the if-clause. then within the if-clause you only need to check what the auth-type cares.